### PR TITLE
use globalStorageUri for user settings

### PR DESCRIPTION
**Pull request by Stunkymonkey**
fix for #1294

this might be a breaking change!?

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
on NixOS I already submitted a patch for this: https://github.com/NixOS/nixpkgs/pull/291021/commits/a00e878cb349df2837f4c746f61aeb5ddf4905ca
So this is already tested by me and others.

**Comment by LasaleFamine** [comment]
Hi @Stunkymonkey thanks for the PR. 
I think this indeed would be a breaking change since `this.userConfigFileUri` is used to load and write the configuration, the old one won't be migrate with only this change. 
You can see the references:
https://github.com/Stunkymonkey/vsc-material-theme/blob/use-globalStorageUri-for-settings/src/core/extension-manager.ts#L121
https://github.com/Stunkymonkey/vsc-material-theme/blob/use-globalStorageUri-for-settings/src/core/extension-manager.ts#L91

I think it would be good to check if the old one is present and if so switch the location to the new one, and the check can be removed when we release a major version. cc @equinusocio

**Comment by Stunkymonkey** [comment]
@LasaleFamine I have no development setup locally, so I am coding blind... without testing it. Be warned.

So I added some more to check for the old file and if existent use it, else use the new one.

Is this what you planned?

**Comment by LasaleFamine** [comment]
@Stunkymonkey actually in this way we are considering the old file as something to use still. If the objective is to migrate to the new use config location, we should copy the old file if present in the new location. 
If the configuration can work either in both locations maybe this change can be ok.

**Comment by LasaleFamine** [comment]
@Stunkymonkey thanks for the change. This feels me the correct way to change the config file. Did you have the possibility to test it?

File: src/core/extension-manager.ts [hunk 1]
```diff
@@ -123,6 +128,23 @@ class ExtensionManager implements IExtensionManager {
...
+      await workspace.fs.rename(oldLocation, newLocation, { overwrite: true });
+      return newLocation;
+    } catch (error) {
+      if (error instanceof FileSystemError) {
+        console.error('Error moving location:', error.message);
+      } else {
+        throw error;
+      }
+      return oldLocation;
+    }
```
Comment (Stunkymonkey) [inline src/core/extension-manager.ts:146]:
```suggestion
    } catch (error) {
      if (error instanceof FileSystemError) {
        console.error('Error moving location:', error.message);
      } else {
        throw error;
      }
    }
    return newLocation;
```

**Comment by Stunkymonkey** [comment]
> @Stunkymonkey thanks for the change. This feels to me the correct way to change the config file. Did you have the possibility to test it?

Sadly no. I do not even have `npm` installed. Is it possible you go on from here?

**Comment by equinusocio** [comment]
There are still typescript error

**Comment by Stunkymonkey** [comment]
could you please take over the PR? I do not have time to setup the dev-env.

**Pull request closed without merging**